pkg/net/net_linux.go

```diff
@@ -163,3 +166,56 @@ func netParseEthtoolFeature(line string) *NICCapability {
 		CanEnable: !fixed,
 	}
 }
+
+func netDeviceHWAddress(netDevDir, netDevName string) NICAddress {
+	nicAddr := NICAddress{}
+
+	// what we do here is not that hard in the end: we need to navigate the sysfs
+	// up to the directory belonging to the device backing the network interface.
+	// we can make few relatively safe assumptions, but the safest way is follow
+	// the right links. And so we go.
+	// First of all, knowing the network device name we need to resolve the backing
+	// device path to its full sysfs path.
+	// say we start with netDevDir="/sys/class/net" and netDevName="enp0s31f6"
+	netPath := filepath.Join(netDevDir, netDevName)
+	dest, err := os.Readlink(netPath)
+	if err != nil {
+		// bail out with empty value
+		return nicAddr
+	}
+	// now we have something like dest="../../devices/pci0000:00/0000:00:1f.6/net/enp0s31f6"
+	// remember the path is relative to netDevDir="/sys/class/net"
```

Comment:
:+1: the comments in this function are exactly the kind of great code comments that `ghw` promotes: explain the details and the *why*. bravo.

Reply:
Thanks!